cupy/_core/include/cupy/carray.cuh

```diff
@@ -199,6 +199,7 @@ __device__ int signbit(float16 x) {return x.signbit();}
          i += static_cast<ptrdiff_t>(blockDim.x) * gridDim.x)
 
 #ifdef CUPY_JIT_MODE
+#include <cupy/swap.cuh>
```

Comment:
ROCm complains about not finding `thrust::swap`, so we have to include it. In fact, I suspect something was wrong on the CUDA call path, because this is a perfectly legitimate complaint.

Reply:
@asi1024 Do you know why CUDA didn't complain about missing `thrust::swap`? 😂

Reply:
It may be related to https://github.com/cupy/cupy/pull/4951.
Before #4951 was merged, JIT test failed on ROCm, but not on CUDA.